File: sorting algorithms/bubble sort/bubble_sort.py
def bubbleSort(arr,size):
    for i in range(size-1):
        swapped=False
        for j in range(size-i-1):
            if arr[j]>arr[j+1]:
                # swapping
                arr[j],arr[j+1]=arr[j+1],arr[j]
                swapped=True
        # if no elements were swapped that means no swapping has taken place
        # list already sorted
        if not swapped:
            break
def printArr(arr):
    for element in arr:
        print(element,end=" ")
    print("")
n = int(input("Enter the size of the list "))
print("\n")
arr = list(int(num) for num in input("Enter elements ").strip().split())[:n]
print("Unsorted List: ")
printArr(arr)
bubbleSort(arr,n)
print("\n")
printArr(arr)

File: sorting algorithms/bubble sort/bubble_sort.c
#include <stdio.h>
#include<stdlib.h>
void bubbleSort(int arr[], int n)
{
    int i, j, temp, swapped=0;
    for(i = 0; i < n; i++)
    {
        for(j = 0; j < n-i-1; j++)
        {
            // introducing a swapped to monitor swapping
            if( arr[j] > arr[j+1])
            {
                // swap the elements
                temp = arr[j];
                arr[j] = arr[j+1];
                arr[j+1] = temp;
                // if swapping happens update swapped to 1
                swapped = 1;
            } 
        }
        // if value of swapped is zero after all the iterations of inner loop
        // then break out
        if(swapped == 0)
        {
            break;
        }
    }
    
    // print the sorted array
    printf("Sorted Array: ");
    for(i = 0; i < n; i++)
    {
        printf("%d  ", arr[i]);
    }
}

int main()
{
    system("cls");
    int arr[100], i, n;
    // ask user for number of elements to be sorted
    printf("Enter the number of elements to be sorted: ");
    scanf("%d", &n);
    // input elements if the array
    for(i = 0; i < n; i++)
    {
        printf("Enter element no. %d: ", i+1);
        scanf("%d", &arr[i]);
    }
    // call the function bubbleSort
    bubbleSort(arr, n);
    
    return 0;
}